### ISSUE TITLE
Help! Not finding any peers 

### ISSUE THREAD

**Issue by ko-redtruck** [holochain/holochain-proto#760]
I'm currently testing my own app locally with two nodes on Windows with Holochain 25. At first I just tried to run two instances on my own computer. I verified that they had the same DNA hash. Then I tried it with a  bootstrap server just like in clutter ReadMe.md. The bootstrap server successfully identified the two nodes: 

`[{"Req":{"Version":1,"NodeID":"QmWQVaqEayZJWnvxLtsKr1iyeTDp3s7m7TTE36HhAUTiTK","NodeAddr":"/ip4/0.0.0.0/tcp/6001"},"Remote":"[::1]:59675","LastSeen":"2018-06-20T17:13:51.0236101+02:00"},
{"Req":{"Version":1,"NodeID":"QmcXesBJFTpgutGnnsqmK1kNVM88Hkn7rQraHousYWJedK","NodeAddr":"/ip4/0.0.0.0/tcp/6002"},"Remote":"[::1]:59671","LastSeen":"2018-06-20T17:13:48.3446662+02:00"}]`

However, it seems that the two nodes still do not see each other. First of all, I can't find anything other than my commits to my local source chain with <code>query()</code> (maybe it is supposed to do that...) but if I try to get a hash this warning appears: <code>DHT Query with no peers in routing table!</code>. 

I tried it with Chrome, Firefox and Opera and non of them work... please help!

my App: https://github.com/ko-redtruck/d.memed

**Comment by ddd-mtl** [comment]
Hi ko-redtruck.
"Remote":"[::1]:59671" shows that its using ipv6.
holochain has problems handling ipv6 correctly on windows for some reason.
Some time ago I did a local hack to fixe this. I'll try to find it and share it to you.

**Comment by ko-redtruck** [comment]
Wow thanks for the quick response! I hope you will find your little hack but otherwise I will test it on a virtual machine

**Comment by ddd-mtl** [comment]
In holochain-proto\cmd\bs.go, add at the beginning of function h, around line 139, add:
`strings.Replace(r.RemoteAddr, "[::1]", "127.0.0.1", -1)`

then recompile bs with:
`go build -a github.com\Holochain\holochain-proto\cmd\bs`

**Comment by ko-redtruck** [comment]
modified and build the file successfully but still can't find any peers

![image](https://user-images.githubusercontent.com/24638508/41673513-7714694e-74bd-11e8-8e5c-fafbf8000e4d.png)

I added the line here

this is now the bootstrap server message: 
`[{"Req":{"Version":1,"NodeID":"QmWQVaqEayZJWnvxLtsKr1iyeTDp3s7m7TTE36HhAUTiTK","NodeAddr":"/ip4/0.0.0.0/tcp/6001"},"Remote":"[::1]:55063","LastSeen":"2018-06-20T19:07:51.1379566+02:00"},{"Req":{"Version":1,"NodeID":"QmcXesBJFTpgutGnnsqmK1kNVM88Hkn7rQraHousYWJedK","NodeAddr":"/ip4/0.0.0.0/tcp/6002"},"Remote":"[::1]:55069","LastSeen":"2018-06-20T19:07:52.1009947+02:00"}]`

**Comment by ddd-mtl** [comment]
You still have `"Remote":"[::1]:55069"` showing, which probably means you are still using the old bs.exe
You should get `"Remote":"127.0.0.1:55069"` instead with the hacked version.

**Comment by ko-redtruck** [comment]
i changed the line to <code> r.RemoteAddr = strings.Replace(r.RemoteAddr, "[::1]", "127.0.0.1", -1)</code>
Now it works! Thanks a lot!!

**Comment by ko-redtruck** [comment]
I now switched to the official bootstrap server: <code>http://bootstrap.holochain.net:10000</code>

Here I can also see my two nodes but it still doesn't work: <code>DHT Query with no peers in routing table!</code> 

`[{"Req":{"Version":1,"NodeID":"12D3KooWL91zXcZumeXddjPer2rhUPakubFiESDWRiMcXsZFZyaH","NodeAddr":"/ip4/0.0.0.0/tcp/61321"},"Remote":"$myIP:59294","LastSeen":"2018-06-21T12:57:33.254536451Z"},{"Req":{"Version":1,"NodeID":"QmQgtrBqu1UJW65DxWHGMVt9o4EuJ9eM4q5DfejEfheADm","NodeAddr":"/ip4/0.0.0.0/tcp/25660"},"Remote":"$myIP:59208","LastSeen":"2018-06-21T13:00:00.960921627Z"}]`

In short: Nodes successfully connected to bootstrap server but still don't see each other.

**Comment by ddd-mtl** [comment]
This one is new to me.
That `$myIP` in "Remote" looks suspicious. Any thoughts @zippy ?

**Comment by ko-redtruck** [comment]
@ddd-mtl Sorry for creating confusion. That's just me being overly cautious and censoring out my ip address.

**Comment by ddd-mtl** [comment]
Is that problem specific to your app?
Have you tried Clutter in the latest release package? You should be able to see a 'Holoport' user.

**Comment by ko-redtruck** [comment]
Clutter (http://bootstrap.holochain.net:10000/QmcUnaykEXWTVbzyWZFQNW4utpuZxEFDb8Tum85pCNX3UC)

`[{"Req":{"Version":1,"NodeID":"QmSakCdrtKKcqtWPK2zUMu9RenPBFrkEFgZKpeZtZJfchz","NodeAddr":"/ip4/0.0.0.0/tcp/6283"},"Remote":"67.176.93.43:60958","LastSeen":"2018-07-15T13:59:00.135936731Z"},{"Req":{"Version":1,"NodeID":"QmZYGTrWvbrf8pPBC5ES4rU5UyTTEBYP5pAbqNxNpnPUSB","NodeAddr":"/ip4/0.0.0.0/tcp/6283"},"Remote":"65.117.15.186:59865","LastSeen":"2018-07-15T14:00:58.738575353Z"},{"Req":{"Version":1,"NodeID":"QmbJM4PEDG36EGeRBWp7K1cLWBPMhvXDM848EbPNMCxg4z","NodeAddr":"/ip4/0.0.0.0/tcp/6283"},"Remote":"63.224.69.150:59780","LastSeen":"2018-07-15T13:59:50.419074855Z"},{"Req":{"Version":1,"NodeID":"QmcWoLQ23bWC3eiX9VTqWBa8mMA26VGQKpfhTwhpNfYkVn","NodeAddr":"/ip4/0.0.0.0/tcp/6283"},"Remote":"37.201.4.158:44450","LastSeen":"2018-07-15T14:01:18.328438553Z"}]`

The last IP is mine but I can't see any other tweets besides mine if that is what you mean

**Comment by marcusnewton** [comment]
I'm getting `"Remote":"127.0.0.1:55069"` for each of my nodes on a local bootstrap with the hack but they still aren't discovering each other

**Comment by Bhanudutta** [comment]
I am using official bootstrap server and having same issue 
There are entries (ipv4 entries) tried with Linux and windows both , but no success 

If I use MDNS it works else it does not 
But MDNS only works on LAN

Does anyone have a solution?
